hello-app/cloudshell/tutorial.md

```diff
@@ -0,0 +1,133 @@
+
+GKE uses Kubernetes objects to create and manage your cluster's resources. This example has two types of Kubernetes objects:
+- a [Deployment](https://cloud.google.com/kubernetes-engine/docs/concepts/deployment) object, which deploys stateless applications like web servers 
+- a [Service](https://cloud.google.com/kubernetes-engine/docs/concepts/service) object, which defines rules and load balancing for accessing your application from the internet
+
+### Create the Deployment
+
+To run `hello-app` in your cluster, run the following command:
+```
+kubectl create deployment hello-server --image=gcr.io/google-samples/hello-app:1.0
+```
+
+Let's break down what this command is doing:
+- The Kubernetes command [`kubectl create deployment`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#create) creates a Deployment named `hello-server`. The Deployment's [Pod](https://cloud.google.com/kubernetes-engine/docs/concepts/pod) runs the `hello-app` container image.
+
+- The command flag `--image` specifies a container image to deploy. In this case, the command pulls the example image from a [Container Registry](https://cloud.google.com/container-registry/docs) bucket, `gcr.io/google-samples/hello-app`. The suffix `:1.0` indicates the specific image version to pull. (If you don't specify a version, the latest version is used.)
+
+## Expose the Deployment
+
+After deploying the application, you need to expose it to the internet so that users can access it. You can expose your application by creating a Service, a Kubernetes resource that exposes your application to external traffic.
+
+To expose your application, run the following [`kubectl expose`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#expose) command:
+```
+kubectl expose deployment hello-server --type LoadBalancer --port 80 --target-port 8080
+```
+
+- Passing in the `--type LoadBalancer` flag creates a Compute Engine load balancer for your container. 
+
+- The `--port` flag initializes public port 80 to the internet and the `--target-port` flag routes the traffic to port 8080 of the application.
+
+Load balancers are billed per Compute Engine's [load balancer pricing](https://cloud.google.com/compute/pricing#lb).
+
+Now let's inspect the Kubernetes resources and access your deployed web app.
+
+## Inspect and view the application
+
+1. Inspect the Pods by running [`kubectl get pods`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#get)
+```bash
+kubectl get pods
+```
+
+You should see one `hello-server` Pod running on your cluster.
+
+2. Inspect the `hello-server` Service by running [`kubectl get service`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#get)
+```bash
+kubectl get service hello-server
+```
+From this command's output, copy the Service's external IP address from the `EXTERNAL-IP` column.
```

Comment:
When I run this command, I see:

```
NAME           TYPE           CLUSTER-IP       EXTERNAL-IP   PORT(S)        AGE
hello-server   LoadBalancer   10.115.250.228   <pending>     80:30069/TCP   65s
```

I wonder if I'm missing a step or if EXTERNAL-IP is just taking a long time to load? Could you make a note about this?

Reply:
Note added!